dice-game-ui-angular/src/app/statistics/statistics.component.ts

```diff
@@ -17,30 +17,38 @@ export interface StatisticsDTO {
   styleUrls: ['./statistics.component.scss'],
 })
 export class StatisticsComponent implements OnInit {
-  statistics: StatisticsDTO[];
+  statistics: StatisticsDTO[] = [];
   displayedColumns: string[];
 
-  constructor(private statisticsService: StatisticsService) {
+  private userId: string;
+
+  constructor(private userService: UserService, private statisticsService: StatisticsService) {
     this.displayedColumns = ['wasCompleted', 'gameType', 'fieldSize', 'turnsCount', 'score', 'isWinner'];
   }
 
   ngOnInit(): void {
-    this.getAllStatistics();
+    this.userService.getUser().subscribe((user) => {
+      this.setUserId(user.id);
```

Comment:
Why do you need to create a separate method for this which is used only once - here?

Reply:
This also leads to misunderstanding, because method name doesn't correspond to what method does.

Reply:
Changed